applications/nrf_desktop/doc/dfu_mcumgr.rst

```diff
@@ -0,0 +1,97 @@
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Device Firmware Upgrade MCUmgr module is responsible for performing a Device Firmware Upgrade (DFU) over Simple Management Protocol (SMP).
+
+You can perform the DFU using, for example, the `nRF Connect for Mobile`_ application.
+The :guilabel:`DFU` button appears in the tab with the connected Bluetooth® devices.
+After pressing the button, you can select the :file:`*.bin` file used for the firmware update.
+
+.. note::
+    The Device Firmware Upgrade MCUmgr module implementation is currently marked as experimental because it uses internal MCUmgr API.
+
+Module events
+*************
+
+.. include:: event_propagation.rst
+    :start-after: table_dfu_mcumgr_start
+    :end-before: table_dfu_mcumgr_end
+
+.. note::
+    |nrf_desktop_module_event_note|
+
+Configuration
+*************
+
+To enable the module, set the :ref:`CONFIG_DESKTOP_DFU_MCUMGR_ENABLE <config_desktop_app_options>` Kconfig option.
+
+The module selects the following configurations:
+
+* :kconfig:option:`CONFIG_MCUMGR` - This option enables the MCUmgr support, which is required for the DFU process.
+  For details, see :ref:`zephyr:mcu_mgr` in the Zephyr documentation.
+* MCUmgr groups:
+
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG` - This option enables the MCUmgr image management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+  * :kconfig:option:`CONFIG_MCUMGR_GRP_OS` - This option enables the MCUmgr OS management handlers, which are required for the DFU process.
+    For details, see :ref:`zephyr:device_mgmt` in the Zephyr documentation.
+
+* :kconfig:option:`CONFIG_MCUMGR_MGMT_NOTIFICATION_HOOKS` - This option enables the MCUmgr notification hook support, which allows the module to listen for an MCUmgr event.
+  For details, see :ref:`zephyr:mcumgr_callbacks` in the Zephyr documentation.
+
+  MCUmgr notifications hooks:
+
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_UPLOAD_CHECK_HOOK` - This option enables the MCUmgr upload check hook, which sends image upload requests to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_IMG_STATUS_HOOKS` - This option enables the MCUmgr image status hooks, which report the DFU status to the registered callbacks.
+    * :kconfig:option:`CONFIG_MCUMGR_GRP_OS_RESET_HOOK` - This option enables the MCUmgr OS reset hook, which sends reset requests to the registered callbacks.
+
+* MCUmgr dependencies:
+
+  * :kconfig:option:`CONFIG_NET_BUF`
+  * :kconfig:option:`CONFIG_ZCBOR`
+  * :kconfig:option:`CONFIG_CRC`
+  * :kconfig:option:`CONFIG_IMG_MANAGER`
+  * :kconfig:option:`CONFIG_STREAM_FLASH`
+  * :kconfig:option:`CONFIG_FLASH_MAP`
+  * :kconfig:option:`CONFIG_FLASH`
+
+To use the module, you must also enable the :kconfig:option:`CONFIG_BOOTLOADER_MCUBOOT` Kconfig option.
+The DFU over Simple Management Protocol in Zephyr is supported only with the MCUboot bootloader.
+
```

Comment:
By default, the MCUmgr DFU module confirms the image calling :c:func:boot_write_img_confirmed during the system boot.